### PR TITLE
refactor: move logic for validateEncryptSubmission to utility

### PR DESCRIPTION
**Pull request by tshuli**

### CODE REVIEW

**Comment by tshuli** [comment]
@karrui thanks for discussing earlier today, could I trouble you to take a look to see if this approach is roughly correct? :)

will clean up / write tests and work on submissions thereafter since you are refactoring the rest of encrypt-submissions

File: src/app/utils/encrypt-submission.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { getProcessedResponses } from '../modules/submission/submission.service'
+
+import { checkIsEncryptedEncoding } from './encryption'
+import { createReqMeta } from './request'
+
+const logger = createLoggerWithLabel(module)
+
+type ReqWithForm<T> = T & { form: IPopulatedForm }
+
+export const isValidEncryptSubmission = (req: Request): boolean => {
```
Comment (karrui) [inline src/app/utils/encrypt-submission.ts:15]: I don't think we need to create the `isValidEncryptSubmission` function, can't we use the `checkIsEncryptedEncoding` function directly?

I think we can update the `checkIsEncryptedEncoding` function into a neverthrown function instead.

Comment (karrui) [inline src/app/utils/encrypt-submission.ts:15]: Utility functions should also not need to be aware of other layers, in this case this function needs to know about `req` from the controller layer, which is not what we want

File: src/app/utils/encrypt-submission.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        formId: form._id,
+      },
+      error,
+    })
+    return false
+  }
+  return true
+}
+
+export const processResponses = (req: Request): void => {
```
Comment (karrui) [inline src/app/utils/encrypt-submission.ts:36]: Same here. The argument for this function should not be `Request`, but scoped to the variable it is going to act on. In this case, this whole block seems to be the domain of the controller and does not need to be extracted.

Also strive to make utility functions [pure](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976). In this function, we are assigning to `req.body.parsedResponses` as a side effect, which can be hard to realize

**Review by karrui — CHANGES_REQUESTED** [review]
I think the things you've refactored out as utilities actually belong in the domain of the controller and does not need to be extracted.

I think what you can do is to convert the utility functions currently being used into neverthrown functions so you don't need to wrap them in a try catch block.

**Comment by tshuli** [comment]
@karrui Thanks a lot! 
Will close this and open a PR for neverthrow refactoring

**Pull request closed without merging**